neherlab/covid19_scenarios
# Run simulation on a webworker

## Related issues and PRs

Contributes to #581

## Description

Runs the simulation algorithm on a webworker.

## Impacted Areas in the application

UI responsiveness improved.
The "Run" button changes to "Running..." while the algorithm runs.

## Testing

~Currently does not run on the staging environment. Likely a production configuration issue(?)~

1. Press 'Run'
  - Observe the button text changes to "Running..."
  - Observe the button text changes back to "Run" when simulation completed.

2. Toggle auto-run
  - Observe the button text changes to "Running..."
  - Observe the button text changes to "Refresh" when simulation completed.

3. Change a scenario parameter
  - Observe the button text changes to "Running..." after ~1 second.
   - Observe the button text changes to "Refresh" when simulation completed.
